internal/provider/config.go

```diff
@@ -118,7 +119,8 @@ func (c *Config) NewClient() (*client.Client, error) {
 	}
 
 	// If there is no cert information, then check for ssh://
-	helper, err := connhelper.GetConnectionHelper(c.Host)
+	ssh_opts_fields := strings.Fields(c.SSHOpts)
+	helper, err := connhelper.GetConnectionHelperWithSSHOpts(c.Host, ssh_opts_fields)
```

Comment:
```suggestion
	helper, err := connhelper.GetConnectionHelperWithSSHOpts(c.Host, strings.Fields(c.SSHOpts))
```

https://golang.org/doc/effective_go#mixed-caps

Reply:
📝 
https://pkg.go.dev/github.com/docker/cli/cli/connhelper#GetConnectionHelper
https://pkg.go.dev/github.com/docker/cli/cli/connhelper#GetConnectionHelperWithSSHOpts